nixos/modules/services/hardware/undervolt.nix

```diff
@@ -0,0 +1,92 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.undervolt;
+in {
+  options.services.undervolt = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to undervolt intel cpus.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.undervolt;
+      defaultText = "pkgs.undervolt";
+      description = ''
+        undervolt derivation to use.
+      '';
+    };
+
```

Comment:
~~(In the following options) it might be better to drop the `Offset` suffix from the option names, pretty redundant and they're all offsets, except for `enable` and `package`.~~ Disregard that as there are more than voltage offsets.

It may be worthwhile to look into supporting other options like `--temp`

Reply:
Good idea! I've added temp support and tested that it works on my NixOS machine